src/main/java/mods/railcraft/common/util/entity/ai/EntityAIMoveToEntity.java

```diff
@@ -0,0 +1,102 @@
+/*------------------------------------------------------------------------------
+ Copyright (c) CovertJaguar, 2011-2018
+ http://railcraft.info
+ 
+ This code is the property of CovertJaguar
+ and may only be used with explicit written
+ permission unless otherwise specified on the
+ license page at http://railcraft.info/wiki/info:license.  
+ -----------------------------------------------------------------------------*/
+package mods.railcraft.common.util.entity.ai;
+
+import mods.railcraft.common.plugins.forge.AIPlugin;
+import mods.railcraft.common.util.entity.EntitySearcher;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityCreature;
+import net.minecraft.entity.ai.EntityAIBase;
+import net.minecraft.entity.ai.RandomPositionGenerator;
+import net.minecraft.util.math.Vec3d;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Predicate;
+
+/**
+ * @author CovertJaguar <http://www.railcraft.info/>
+ */
+public class EntityAIMoveToEntity extends EntityAIBase {
```

Comment:
Rename to EntityAISearchForEntity, and the block version to EntityAISearchForBlock.